app/test_main.py

```diff
@@ -1 +1,41 @@
-# write your code here
+import pytest
+from unittest import mock
+from app.main import can_access_google_page
+from mock import MagicMock
+
+
+@pytest.fixture
+def mock_valid_google_url() -> MagicMock:
+    with (mock.patch("app.main.valid_google_url")
+          as mocked_valid_google_url):
+        yield mocked_valid_google_url
+
+
+@pytest.fixture
+def mock_has_internet_connection() -> MagicMock:
```

Comment:
it would make sense to put mocks in a fixture if you have many test and you are going to reuse them. but since you have only one test why you use fixtures here?